### PR TITLE
[MIG] partner_changeset: Migrated to 10.0

### PR DESCRIPTION
**Pull request by yvaucher**
History taken from 9.0 branch

for migration change diff see commit https://github.com/OCA/partner-contact/pull/380/commits/81398af2c5b97fc426f1c1a409ceb89d3daae6ca

### CODE REVIEW

**Comment by yvaucher** [comment]
travis error doesn't seems to be related, this might be an upstream bug (Access error when installing base xml data)

restarted the job

File: partner_changeset/README.rst
Comment (cubells) [inline partner_changeset/README.rst]: It must be 10.0

File: partner_changeset/models/changeset_field_rule.py
Comment (cubells) [inline partner_changeset/models/changeset_field_rule.py]: Reorder imports please: from odoo import api, fields, models

File: partner_changeset/models/res_partner.py
Comment (cubells) [inline partner_changeset/models/res_partner.py]: SAme as above

File: partner_changeset/models/res_partner.py
Comment (cubells) [inline partner_changeset/models/res_partner.py:17]: Must start with _compute

File: partner_changeset/models/res_partner.py
Comment (cubells) [inline partner_changeset/models/res_partner.py]: use of api.one is not recommended

File: partner_changeset/models/res_partner_changeset.py
Comment (cubells) [inline partner_changeset/models/res_partner_changeset.py]: same as above

File: partner_changeset/models/res_partner_changeset.py
Comment (cubells) [inline partner_changeset/models/res_partner_changeset.py]: use of api.one is not recommended

File: partner_changeset/models/res_partner_changeset.py
Comment (cubells) [inline partner_changeset/models/res_partner_changeset.py]: use of api.one is not recommended

File: partner_changeset/models/res_partner_changeset.py
Comment (cubells) [inline partner_changeset/models/res_partner_changeset.py]: use of api.one is not recommended

**Review by cubells — CHANGES_REQUESTED** [review]
Fast code review

**Comment by cubells** [comment]
Travis error is not related to this module.

**Review by antoniocanovas — CHANGES_REQUESTED** [review]
Work as described.
It would be good to have any signal for the end user with the pending approvals (like admin user).
If it is not implemented the end user will do the changes several times and finally will create an issue.

**Review by leemannd — APPROVED** [review]
Thanks. Just squash transifex commits

**Comment by damdam-s** [comment]
@yvaucher @leemannd @cubells can you please check it to have this one merged very soon please ?

**Comment by damdam-s** [comment]
tests fails because of this module `partner_create_by_vat`

File: partner_changeset/models/res_partner_changeset.py
Comment (rousseldenis) [inline partner_changeset/models/res_partner_changeset.py]: This must be 'model'

File: partner_changeset/models/res_partner_changeset.py
Comment (rousseldenis) [inline partner_changeset/models/res_partner_changeset.py]: I'm not confident in this as you create new cache at each loop iteration.

Maybe a better approach should be a other 'write' method on partner

File: partner_changeset/models/res_partner_changeset.py
Comment (rousseldenis) [inline partner_changeset/models/res_partner_changeset.py]: This must be model

File: partner_changeset/models/res_partner_changeset.py
Comment (rousseldenis) [inline partner_changeset/models/res_partner_changeset.py]: api.model

**Review by rousseldenis — CHANGES_REQUESTED** [review]
LGTM. Code Review.

Some comments

**Comment by damdam-s** [comment]
@pedrobaeza @cubells @rousseldenis fixes done. can we go forward on this ?

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

Code review and test.

Good job @damdam-s

**Review by rousseldenis — APPROVED** [review]
Code review :+1: 

@damdam-s Can you see why travis tests fail?

**Comment by cubells** [comment]
@rousseldenis @damdam-s 
Travis fails due to `partner_create_by_vat` tests.

As `partner_create_by_vat` is executed after `partner_changeset`, and `partner_changeset` tests change permission of street field, when `partner_create_by_vat` test try to change street value, this value is not changed.

So @damdam-s you must restore partner fields permissions in tests in order to avoid similar fails in future.

Do you understand?

**Comment by damdam-s** [comment]
@cubells tests corrected. 
Is this PR ok for you ?
Thanks for your time

**Comment by cubells** [comment]
@damdam-s 

I'll retest later.

**Comment by damdam-s** [comment]
@tafaRU @eLBati @pedrobaeza as PSC members, can you review this PR please :smile: ? many thanks

**Review by leemannd — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Can you squash last 3 commits before merging?

**Comment by damdam-s** [comment]
@pedrobaeza of course. done
Thanks for taking care :smiley:

**Pull request merged**
